tests/apollo/test_skvbc.py

```diff
@@ -57,13 +58,16 @@ async def test_state_transfer(self, bft_network):
         cluster with f=1 we should be able to stop a different node after state
         transfer completes and still operate correctly.
         """
+        await bft_network.do_key_exchange()
```

Comment:
shouldn't the replicas be started before we do the key exchange?

Reply:
This method starts the replicas, performs key exchange, and stops the replicas.
The idea is to add this method as an initial step to an arbitrary test. Then the test starts as it used to start, for example, if a test needs to start with a partial set of the replicas, this method won't impose any changes.

The test can test its functionality with the insurance that it uses an updated start-up sequence but without any changes to the test itself.

Of course, it does create additional side effects that can't be avoided, like starting from a last executed sequence number which is greater than 0.

Reply:
I see, thanks Eran. I would then suggest the following improvement:
* add an optional parameter to the decorator `def with_bft_network(await bft_network.do_key_exchange(), ..., rotate_keys=False)`
* move the call to `await bft_network.do_key_exchange()` in the `with_bft_network` decorator. Only run `if rotate_keys`.
* update the decorator for tests where you want to do key rotation (e.g. `with_bft_network(start_replica_cmd, rotate_keys=True)`)

Reply:
It's a great improvement, I hope I'll be able to do it for this PR, since I'm on a very tight schedule.
If not I'll open a ticket and implement it after this.

Reply:
you wrote ```def with_bft_network(await bft_network.do_key_exchange(), ..., rotate_keys=False)```
do I need the ```await bft_network.do_key_exchange()``` as a param ?

Reply:
Oops, of course not, it was a copy/paste glitch ;) just add the `rotate_keys` param. And sure, this improvement can be in a separate PR!